src/WindowsTracing/ContextSwitchManagerTest.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "ContextSwitchManager.h"
+#include "MockTracerListener.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+
+namespace orbit_windows_tracing {
+
+TEST(ContextSwitch, ListenerIsCalled) {
+  orbit_windows_tracing::MockTracerListener mock_listener;
+  ContextSwitchManager manager(&mock_listener);
+
+  EXPECT_CALL(mock_listener, OnSchedulingSlice).Times(testing::Exactly(1));
```

Comment:
Nit: you can simply state `Times(1)`.
Also, I would also add do a `SaveArg`, to check that the scheduling slice contains the correct information.

Both applies for here and below.

Reply:
Edit (since I read the code below and the Mock class):
Ok, I see using the DelegateToFake. I think that is fine. `SaveArg` would be probably my choice here, but that's personal preference.

However, I'd do the actual check for the values of the scheduling slice in this test anyways. Or just remove this test case, as it does not gain any more information than the one next.

Reply:
The pattern comes from here http://google.github.io/googletest/gmock_cook_book.html#DelegatingToFake. Will see if SaveArg is simpler.

Reply:
Removed `FakeTracerListener` in favor of SaveArg.